src/tlo/methods/cervical_cancer.py

```diff
@@ -0,0 +1,1542 @@
+            Types.BOOL,
+            "ever been treated for cc"
+        ),
+        "ce_cc_ever": Property(
+            Types.BOOL,
+            "ever had cc"
+        ),
+            # currently this property has levels to match ce_hov_cc_status to enable the code as written, even
+            # though can only be treated when in stage 1-3
+        "ce_stage_at_which_treatment_given": Property(
+            Types.CATEGORICAL,
+            "the cancer stage at which treatment was given (because the treatment only has an effect during the stage"
+            "at which it is given).",
+            categories=["none", "hpv", "cin1", "cin2", "cin3", "stage1", "stage2a", "stage2b", "stage3", "stage4"],
+        ),
+        "ce_date_palliative_care": Property(
+            Types.DATE,
+            "date of first receiving palliative care (pd.NaT is never had palliative care)"
+        ),
+        "ce_ever_diagnosed": Property(
+            Types.DATE,
+            "ever diagnosed with cervical cancer (even if now cured)"
+        ),
+        "ce_date_death": Property(
+            Types.DATE,
+            "date of cervical cancer death"
+        ),
+        "ce_new_stage_this_month": Property(
+            Types.BOOL,
+            "new_stage_this month"
+        ),
+        "ce_xpert_hpv_ever_pos": Property(
+            Types.BOOL,
+            "hpv positive on xpert test ever"
+        ),
+        "ce_via_cin_ever_detected": Property(
+            Types.BOOL,
+        "cin ever_detected on via"
+        ),
+        "ce_date_cryo": Property(
+            Types.DATE,
+        "date of cryotherapy for CIN"
+        ),
+        "ce_current_cc_diagnosed": Property(
+            Types.BOOL,
+            "currently has diagnosed cervical cancer (which until now has not been cured)"
+        ),
+        "ce_selected_for_via_this_month": Property(
+            Types.BOOL,
+            "selected for via this period"
+        ),
+        "ce_selected_for_xpert_this_month": Property(
+            Types.BOOL,
+            "selected for xpert this month"
+        ),
+        "ce_biopsy": Property(
+            Types.BOOL,
+            "ce biopsy done"
+        )
+    }
+
+    def read_parameters(self, data_folder):
+        """Setup parameters used by the module, now including disability weights"""
+
+        # Update parameters from the resourcefile
+        self.load_parameters_from_dataframe(
+            pd.read_excel(Path(self.resourcefilepath) / "ResourceFile_Cervical_Cancer.xlsx",
+                          sheet_name="parameter_values")
+        )
+
+        # note that health seeking probability quite high even though or =1
+        self.sim.modules['SymptomManager'].register_symptom(
+            Symptom(name='vaginal_bleeding',
+                    odds_ratio_health_seeking_in_adults=1.00)
+        )
+
+        # in order to implement screening for cervical cancer creating a dummy symptom - likely there is a better way
```

Comment:
The comment is right here -- this is having the desired effect, but we would rather not use the 'symptoms' in this way.

Instead, for the individuals that are chosen to be screened, create and schedule the HSI event directly. 

e.g. for each indivisual to be screened... make an HSI_Event_CervicalCancer_Screening..... and in that event, do whatever is required for the sceening. (might be the same as happens in the generic appointment, in which case point them both to the same function)

Reply:
Thanks - I will see if I can make that work.